packages/std/src/addresses.rs

```diff
@@ -138,24 +138,96 @@ impl<'a> From<&'a Addr> for Cow<'a, Addr> {
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq, Hash, JsonSchema)]
 pub struct CanonicalAddr(pub Binary);
 
+/// Implement `CanonicalAddr == Binary`
+impl PartialEq<Binary> for CanonicalAddr {
+    fn eq(&self, rhs: &Binary) -> bool {
+        &self.0 == rhs
+    }
+}
+
+/// Implement `Binary == CanonicalAddr`
+impl PartialEq<CanonicalAddr> for Binary {
+    fn eq(&self, rhs: &CanonicalAddr) -> bool {
+        self == &rhs.0
+    }
+}
+
+/// Implement `CanonicalAddr == HexBinary`
+impl PartialEq<HexBinary> for CanonicalAddr {
+    fn eq(&self, rhs: &HexBinary) -> bool {
+        self.as_slice() == rhs.as_slice()
+    }
+}
+
+/// Implement `HexBinary == CanonicalAddr`
+impl PartialEq<CanonicalAddr> for HexBinary {
+    fn eq(&self, rhs: &CanonicalAddr) -> bool {
+        self.as_slice() == rhs.0.as_slice()
+    }
+}
+
 impl From<&[u8]> for CanonicalAddr {
     fn from(source: &[u8]) -> Self {
         Self(source.into())
     }
 }
 
+// Array reference
+impl<const LENGTH: usize> From<&[u8; LENGTH]> for CanonicalAddr {
+    fn from(source: &[u8; LENGTH]) -> Self {
+        Self(source.into())
+    }
+}
```

Comment:
What about a slice?

Reply:
You mean `impl From<&[u8]> for CanonicalAddr`? This was already implemented before a few lines above this addition.

Reply:
Ah, my bad!